smallvil/src/handlers/xdg_shell.rs

```diff
@@ -0,0 +1,125 @@
+use smithay::{
+    delegate_xdg_shell,
+    desktop::{Kind, Window, WindowSurfaceType},
+    reexports::{
+        wayland_protocols::xdg::shell::server::xdg_toplevel,
+        wayland_server::{protocol::wl_surface::WlSurface, DisplayHandle, Resource},
+    },
+    utils::Rectangle,
+    wayland::{
+        seat::{PointerGrabStartData, Seat},
+        shell::xdg::{XdgRequest, XdgShellHandler, XdgShellState},
+        Serial,
+    },
+};
+
+use crate::{
+    grabs::{MoveSurfaceGrab, ResizeSurfaceGrab},
+    Smallvil,
+};
+
+impl XdgShellHandler for Smallvil {
+    fn xdg_shell_state(&mut self) -> &mut XdgShellState {
+        &mut self.xdg_shell_state
+    }
+
+    fn request(&mut self, _dh: &DisplayHandle, request: XdgRequest) {
+        match request {
+            XdgRequest::NewToplevel { surface } => {
+                let window = Window::new(Kind::Xdg(surface.clone()));
+                self.space.map_window(&window, (0, 0), false);
+
+                surface.send_configure();
```

Comment:
The initial configure should be sent after the inital commit. I can not remember which client it was, but there was definitely one where sending the configure event before the initial commit caused problems.

> After creating a role-specific object and setting it up, the client must perform an initial commit without any buffer attached. The compositor will reply with an xdg_surface.configure event.